ci/conda_env_python.yml

```diff
@@ -26,4 +26,3 @@ pytest-lazy-fixture
 pytz
 setuptools
 setuptools_scm=3.2.0
-pickle5
```

Comment:
It would be nice to install `pickle5` in the Python 3.6 worker.

Reply:
@pitrou may we handle that as a follow-up?

Reply:
As you prefer, though it's probably a single bash conditional.

Reply:
I'd like to merge it to move forward with the dependent PR in order to prevent queueing GHA.

Created a JIRA to track it https://issues.apache.org/jira/browse/ARROW-7219